include/eal.h

```diff
@@ -5,41 +5,46 @@
 #include <string>
 #include <vector>
 
+#include <gsl/gsl_interp.h>
+
 namespace eal {
 
-  enum interpolation {
-    linear,
-    spline
+  enum derivation {
+    none,
+    first,
+    second
   };
```

Comment:
Enum for derivations might be unnecessary? Just use numbers directly?

Reply:
I think using numbers is fine. 0, 1, 2, ...